File: src/main/java/com/example/petsrego/model/Pet.java
package com.example.petsrego.model;

import java.util.UUID;
import org.springframework.cloud.gcp.data.spanner.core.mapping.PrimaryKey;
import org.springframework.cloud.gcp.data.spanner.core.mapping.Table;

@Table(name = "pet")
public class Pet {
  @PrimaryKey
  private String id;
  private String name;
  private String type;
  private String breed;
  private String picture;
  private String owner;

  public Pet() {
    this.id = UUID.randomUUID().toString();
  }

  public String getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getType() {
    return type;
  }

  public void setType(String type) {
    this.type = type;
  }

  public String getBreed() {
    return breed;
  }

  public void setBreed(String breed) {
    this.breed = breed;
  }

  public String getPicture() {
    return picture;
  }

  public void setPicture(String picture) {
    this.picture = picture;
  }

  public String getOwner() {
    return owner;
  }

  public void setOwner(String owner) {
    this.owner = owner;
  }
}

File: src/main/kotlin/com/example/petsrego/model/Pet.kt
package com.example.petsrego.model

import org.springframework.cloud.gcp.data.spanner.core.mapping.PrimaryKey
import org.springframework.cloud.gcp.data.spanner.core.mapping.Table
import java.util.*

@Table(name = "pet")
data class Pet(
        @PrimaryKey
        var id: String = UUID.randomUUID().toString(),
        var name: String? = null,
        var type: String? = null,
        var breed: String? = null,
        var picture: String? = null,
        var owner: String? = null)